### DISCUSSION TITLE
Few questions before getting started

### DISCUSSION THREAD

**Discussion by JRBros2346** [gluesql/gluesql#1509] [Q&A] [answered]
Is this like a full fledged RDBMS?
Can it be used as embedded Database like Sqlite?
Does it support atleast Few Relational Queries? (I need a lot)
Does it support Datetime primitives?
Is ASTBuilder mature enough to create any queries?

**Comment by devgony** [comment, marked as answer]
> Is this like a full fledged RDBMS?

I can say GlueSQL is full fledged RDBMS which covers most of ANSI SQL and support various features like dump, import, CLI, ..
You can see details here
https://gluesql.org/docs/0.15/
Or if there is missing features you want, create issue. Then we will comment whether it is supported or the estimated date in roadmap

> Can it be used as embedded Database like Sqlite?

Sure, basically most of GlueSQL's storages are embedded, except for Mongo and Redis storages which support network server as well.

> Does it support at least Few Relational Queries? (I need a lot)

Currently, SQL supports INNER and LEFT OUTER JOIN.
The planner supports HASH and NESTED LOOP JOIN
Foreign key(#1400) is supported as well (just merged into main few days ago, not released yet)

> Does it support Datetime primitives?

GlueSQL supports chrono::{NaiveDate, NaiveDateTime, NaiveTime}.
refer to here
https://gluesql.org/docs/0.15/sql-syntax/data-types/date
https://gluesql.org/docs/0.15/sql-syntax/data-types/timestamp
https://gluesql.org/docs/0.15/sql-syntax/data-types/time

> Is ASTBuilder mature enough to create any queries?

Our AST builder covers all queries GlueSQL supports and even more.
Which means, If you want to handle plan manually like `Hash Join` it is possible by AST builder

**Reply by JRBros2346** [reply]
I also happen to use a lot of Money types. It seems Decimal type is a nice choice, but what is the underlying rust type?

EDIT: nvm i found it. Seems like GlueSQL is what i was looking for :)

**Reply by JRBros2346** [reply]
Well, how to use Decimal type in insert queries?
```rust
table("moneys")
    .insert()
    .values(vec![vec![/*What goes here?*/]])
    .execute(glue)
    .await?;
```
